owlkettle/adw.nim

```diff
@@ -1286,6 +1289,141 @@ renderable AboutWindow {.since: AdwVersion >= (1, 2).}:
+  timeout: int = 5, 
+  useMarkup: bool = false,
+  customTitle: Widget = nil.Widget
+): Toast =
+  result = Toast(
+    title: title, 
+    buttonLabel: buttonLabel,
+    priority: priority,
+    timeout: timeout,
+    dismissalHandler: dismissalHandler
+  )
+  
+  when AdwVersion >= (1, 2):
+    result.clickedHandler = clickedHandler
+    result.customTitle = customTitle
+  else:
+    let isUsingCustomTitle = not customTitle.isNil()
+    if isUsingCustomTitle:
+      raise newException(LibraryError, "The customTitle field on a Toast instance is not available when compiling for Adwaita versions below 1.2. Compile for Adwaita version 1.2 or higher with -d:adwminor=2 to enable it")
+
+    let isUsingClickedHandler = not clickedHandler.isNil()
+    if isUsingClickedHandler:
+      raise newException(LibraryError, "The clickedHandler field on a Toast instance is not available when compiling for Adwaita versions below 1.2. Compile for Adwaita version 1.2 or higher with -d:adwminor=2 to enable it")
+      
+  when AdwVersion >= (1, 4):
+    result.useMarkup = useMarkup
+  else:
+    let isUsingUseMarkup = useMarkup == true
+    if isUsingUseMarkup:
+      raise newException(LibraryError, "The useMarkup field on a Toast instance is not available when compiling for Adwaita versions below 1.4. Compile for Adwaita version 1.4 or higher with -d:adwminor=4 to enable it")
+
+proc connectSignal(obj: pointer, userCallback: proc() {.closure.}, eventName: string) =
+  proc callback(obj: pointer, data: ptr EventObj[proc ()]) {.cdecl.} = 
+    let event = unwrapSharedCell(data)
+    event.callback()
+    # Disconnect event-handler after Toast was dismissed
+    g_signal_handler_disconnect(obj, event.handler)
+  
+  let event = EventObj[proc()]()
+  let data = allocSharedCell(event)
+  data.callback = userCallback
+  data.handler = g_signal_connect(obj, eventName.cstring, callback, data)
+
+proc toGtk(toast: Toast): AdwToast =
+  result = adw_toast_new(toast.title.cstring)
+  if toast.buttonLabel != "":
+    adw_toast_set_button_label(result, toast.buttonLabel.cstring)
+  adw_toast_set_priority(result, toast.priority)
+  adw_toast_set_timeout(result, toast.timeout.cuint)
+  
+  # Set Dismissal Handler
+  if not toast.dismissalHandler.isNil():
+    connectSignal(pointer(result), toast.dismissalHandler, "dismissed")
+  
+  when AdwVersion >= (1, 2):
+    if not toast.customTitle.isNil():
+      let customTitleWidget = toast.customTitle.build().unwrapInternalWidget()
+      adw_toast_set_custom_title(result, customTitleWidget)
+
+    # Set Clicked Handler
+    if not toast.clickedHandler.isNil():
+      connectSignal(pointer(result), toast.clickedHandler, "button-clicked")
+
+  when AdwVersion >= (1, 4):
+    adw_toast_set_use_markup(result, toast.useMarkup.cbool)
+
+type ToastQueue* = ref object
+  toasts: seq[Toast]
+  addToOverlay: proc(toast: Toast) {.closure.}
+
+proc isInitialized(queue: ToastQueue): bool = not queue.addToOverlay.isNil()
```

Comment:
Unused

Reply:
Ah, yeah, the entire "addToOverlay" part was from an earlier attempt at getting this to work.
Given it works without that now I summarily removed it